OpenRA.Mods.Common/Traits/World/WeatherOverlay.cs

```diff
@@ -69,10 +69,19 @@ public class WeatherOverlayInfo : TraitInfo, ILobbyCustomRulesIgnore
 		[Desc("Works only with line enabled and can be used to fade out the tail of the line like a contrail.")]
 		public readonly byte LineTailAlphaValue = 200;
 
+		[Desc("How long the time the weather fades in when enabled.")]
+		public readonly int FadeOutTicks = 1000;
+
+		[Desc("How long the time the weather fades out when disabled.")]
+		public readonly int FadeInTicks = 1000;
+
+		[Desc("Percentage of the initial particle when enabled and the game start.")]
+		public readonly int InitialParticlePercentage = 100;
```

Comment:
This feels a bit speculatively general IMO. Do we have a realistic usecase for needing a value that isn't 100 when it starts enabled and 0 when it starts disabled?

Reply:
This option is used for when player want to start from mid-stage of the weather, any weather. For example, a map starts from mid-stage of a thundershower: there is only mid rain but get heavier as time goes by, when it at max stage there will be lightning and thunder by using `FlashPostProcessEffect` (we can make it conditional and scriptable later).